base/src/Data/Macaw/CFG/Rewriter.hs

```diff
@@ -507,13 +507,17 @@ rewriteApp app = do
     -- (x << j) testBit i ~> x testBit (i-j)
     -- plus a couple special cases for when the tested bit falls outside the shifted value
     BVTestBit (valueAsApp -> Just (BVShr w x (BVValue _ j))) (BVValue _ i)
-      | j + i <= maxUnsigned w -> do
+      | j + i < intValue w ->
       rewriteApp (BVTestBit x (BVValue w (j + i)))
+      | otherwise -> pure (boolLitValue False)
     BVTestBit (valueAsApp -> Just (BVSar w x (BVValue _ j))) (BVValue _ i)
-      | i < intValue w -> do
-      rewriteApp (BVTestBit x (BVValue w (min (j + i) (intValue w-1))))
+      | j + i < intValue w ->
+      rewriteApp (BVTestBit x (BVValue w (j + i)))
+      | i < intValue w -> pure (boolLitValue True)
+      | otherwise -> pure (boolLitValue False)
```

Comment:
I think these two cases aren't quite right: they should both pick out the top bit of x.

Reply:
Or, hm. Maybe the first one should pick out the top bit and the second one is fine? I don't know what the intended behavior of testBit for out-of-range indices is.

Reply:
You are right -- the second one should pick out the top bit. Let me see if I can fix it.